src/NLU.DevOps.Luis/LuisNLUTestClient.cs

```diff
@@ -140,17 +138,17 @@ Entity getEntity(EntityModel entity)
                     entityScore = scoreValue;
                 }
 
-                return entityScore.HasValue
-                    ? new ScoredEntity(entityType, entityValue, matchText, matchIndex, entityScore.Value)
-                    : new Entity(entityType, entityValue, matchText, matchIndex);
+                return new Entity(entityType, entityValue, matchText, matchIndex)
+                    .WithScore(entityScore);
             }
 
-            var intent = speechLuisResult.LuisResult.TopScoringIntent?.Intent;
-            var score = speechLuisResult.LuisResult.TopScoringIntent?.Score;
-            var entities = speechLuisResult.LuisResult.Entities?.Select(getEntity).ToList();
-            return !score.HasValue && Math.Abs(speechLuisResult.TextScore) < Epsilon
-                ? new LabeledUtterance(speechLuisResult.LuisResult.Query, intent, entities)
-                : new ScoredLabeledUtterance(speechLuisResult.LuisResult.Query, intent, score ?? 0, speechLuisResult.TextScore, entities);
+            return new LabeledUtterance(
+                    speechLuisResult.LuisResult.Query,
+                    speechLuisResult.LuisResult.TopScoringIntent?.Intent,
+                    speechLuisResult.LuisResult.Entities?.Select(getEntity).ToList())
+                .WithScore(speechLuisResult.LuisResult.TopScoringIntent?.Score)
+                .WithTextScore(speechLuisResult.TextScore)
+                .WithTimestamp(DateTimeOffset.Now);
```

Comment:
>.WithTimestamp(DateTimeOffset.Now); [](start = 16, length = 35)

does it have to be attached to each utterance?  We could have a timestamp at the beginning of the json file for all utterances

Reply:
Worried about long running tests that take O(1hr) where you may have trouble with hours / minutes in the future (I.e., book me a car in 15 minutes)

Reply:
Won't fix for now, we can optimize later.

Reply:
We may want to consider making the timestamp opt-in though...

Reply:
(will send a follow up PR).